### PR TITLE
minor fix in jax attention bias

### LINKED ISSUE #19897: Flax implementation of BART contain NaN in hidden_states
### System Info

While implementing Donut model #19831, one of my test was failing `test_from_pretrained_save_pretrained`, while debugging the testcase failure, It is happening because the test is trying to compare nan with nan which causes the failure, on tracing the root cause, it came down to this line [`jnp.full(attention_mask.shape, float("-inf")).astype(self.dtype)`](https://github.com/huggingface/transformers/blob/fdffee8a601d0408c7e0f57fbb56217f8b57e62a/src/transformers/models/mbart/modeling_flax_mbart.py#L386), `float("-inf")` is causing the `dot_product_attention_weights` to return NaN instead of 0, which is getting cascaded downstream. Since this code is copied from BART, and that code has been copied to several different models (OPT, PEGASUS, BLENDERBOT etc), I am raising this issue against BART

IMHO, we should replace `float("-inf")` with `-1e10` as is the case for multiple different model, RoBerta, if the maintainers agree with this understanding and solution, I can raise a quick PR to fix the same, or otherwise please suggest a solution

@patil-suraj @patrickvonplaten 

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Run testcase from my branch in the above mentioned PR

### Expected behavior

Hidden States should contain 0 instead of instead of NaN

### PR DESCRIPTION
**Pull request by amankhandelia**
Fixes #19897 

Adding this PR to check if this passes all the test cases, or fix have potential issues

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sanchit-gandhi** [comment]
Hey @amankhandelia! In general, I'm happy with this change. What worries me is that the tests for BART and BART-dervied models currently pass on main, which suggests there shouldn't be a need to change the attention mask value. It suggests that there could be an issue with the FlaxMBartForCausalLM model that you're adding. I've replied more in-depth on the issue as it's more relevant there https://github.com/huggingface/transformers/issues/19897#issuecomment-1294648919. Keeping this PR open until we determine whether it's a generic Flax BART issue or a FlaxBartForCausalLM one!
